addons/calendar/static/src/views/calendar_form/calendar_form_controller.js

```diff
@@ -40,4 +41,13 @@ export class CalendarFormController extends FormController {
         }
         return super.beforeExecuteActionButton(...arguments);
     }
+
+    async deleteRecord() {
+        const action = await this.orm.call("calendar.event", "action_unlink_event", [
+            this.model.root.resId,
+            this.model.root.data.partner_ids.resIds,
+            this.model.root.data.recurrence_update,
+        ]);
+        this.actionService.doAction(action);
```

Comment:
Hello :wave:
Is there a specific reason why this call is asynchronous and this one is not?
https://github.com/odoo/odoo/pull/170027/files#diff-d9b4c3e27348d4faba70e799caec916ad47b3e14bf644d11a1302ad3682efd35R93

Reply:
Hello @geflx 
the diff you mentioned is a method deleteRecord() which is not async in the parent(CalenderController component) itself thus we can not make async it while extend it on to other hand here in formcontroller deleteRecord() is async in FormController in parent so that is the reason. 

I've made the changes you suggested.
Thank you Have a nice weekend :slightly_smiling_face: